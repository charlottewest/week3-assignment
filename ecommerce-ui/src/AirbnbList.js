import React from 'react';
import PropTypes from 'prop-types';
import Airbnb from './Airbnb';
import './Airbnb.css';

export default class AirbnbList extends React.Component {
  static propTypes = {
    airbnbs: PropTypes.array
  }

  render() {
    const airbnbList = this.props.airbnbs
      .map((airbnb, index) => {
        return (
          <Airbnb
            airbnb={airbnb}
            onClick={this.props.onbookAirbnb}
            index={index}
            key={index}
          />
          );
      });

    return (
      <div>
        <div className='airbnbSection'>
          {airbnbList}
        </div>
      </div>
    );
  }
}
