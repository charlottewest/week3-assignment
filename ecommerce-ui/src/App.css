div.container {
  padding: 80px;
}

h1.airbnbSectionHeader {
  font-size: 20px;
  text-decoration: underline;
}

div.shoppingCartContainer {
  padding: 50px;
}

div.availablePropertiesContainer {
  padding: 50px;
}
