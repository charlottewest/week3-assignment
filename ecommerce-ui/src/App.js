import React from 'react';
import PropTypes from 'prop-types';
import ShoppingCart from './ShoppingCart';
import Airbnb from './Airbnb';
import AirbnbList from './AirbnbList';
import Airbnbs from './airbnbs.json';
import './App.css';
import './Airbnb.css';

class App extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      airbnbs: Airbnbs,
      selectedAirbnbs: [],
      totalPayment: 0
    }
  }

  bookAirbnb = (index) => {
    const selectedAirbnb = this.state.airbnbs[index];
    if (this.state.selectedAirbnbs.includes(selectedAirbnb)) {
      alert('You already selected that listing.');
      return;
    }

    const newPayment = this.state.totalPayment + selectedAirbnb.payment.cost;

    this.setState(prevState => {
      return {
        selectedAirbnbs: [...prevState.selectedAirbnbs, selectedAirbnb],
        totalPayment: newPayment
      };
    });
  }

  deleteAirbnb = (index) => {
    this.setState(prevState => {
      const selectedAirbnb = this.state.selectedAirbnbs[index];
      const newSelectedAirbnbs = [...this.state.selectedAirbnbs];
      newSelectedAirbnbs.splice(index, 1);
      const newPayment = this.state.totalPayment - selectedAirbnb.payment.cost;

      return {
        selectedAirbnbs: newSelectedAirbnbs,
        totalPayment: newPayment
      };
    });
  }

  render() {
    return (
      <div className='App'>
        <div className='shoppingCartContainer'>
          <h2>Shopping Cart</h2>
          <ShoppingCart
            airbnbs={this.state.selectedAirbnbs}
            ondeleteAirbnb={this.deleteAirbnb}
            totalPayment={this.state.totalPayment}
          />
        </div>
        <hr />
        <div className='availablePropertiesContainer'>
          <h2>Available Properties</h2>
          <AirbnbList
            airbnbs={this.state.airbnbs}
            onbookAirbnb={this.bookAirbnb}
          />
        </div>
      </div>
    );
  }
}

export default App;
