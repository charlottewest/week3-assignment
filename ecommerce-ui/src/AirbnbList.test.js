import React from 'react';
import ReactDOM from 'react-dom';
import {shallow, render, mount} from 'enzyme';
import App from './App';
import AirbnbList from './AirbnbList';
import Airbnbs from './airbnbs.json';

it('shallow renders', () => {
  const wrapper = shallow(<AirbnbList airbnbs={Airbnbs}/>);
  expect(wrapper.render()).toMatchSnapshot();

  expect(wrapper.props('children')).toBeDefined();
});
