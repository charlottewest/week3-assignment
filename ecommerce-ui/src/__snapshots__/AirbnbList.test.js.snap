// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow renders 1`] = `
<div>
  <div
    class="airbnbSection"
  >
    <div
      class="airbnb-grid"
    >
      <img
        class="airbnbImg"
        src="https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80"
      />
      <span
        class="airbnbTitle"
      >
        Centrally-located Manhattan studio
      </span>
      <span
        class="airbnbCost"
      >
        $105
      </span>
      <button
        class="airbnbBookingButton"
      >
        Book
      </button>
    </div>
    <div
      class="airbnb-grid"
    >
      <img
        class="airbnbImg"
        src="https://images.unsplash.com/photo-1519424187720-db6d0fc5a5d2?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=19f783ae564d7c6da4777cde1a84c1c9&auto=format&fit=crop&w=500&q=80"
      />
      <span
        class="airbnbTitle"
      >
        Denver apartment outside Union Station
      </span>
      <span
        class="airbnbCost"
      >
        $130
      </span>
      <button
        class="airbnbBookingButton"
      >
        Book
      </button>
    </div>
    <div
      class="airbnb-grid"
    >
      <img
        class="airbnbImg"
        src="https://images.unsplash.com/photo-1495136568018-11a54caad6b5?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=b4d19de84a578b1175e85e0811f41c97&auto=format&fit=crop&w=500&q=80"
      />
      <span
        class="airbnbTitle"
      >
        Quaint 3-bedroom home in Gamla stan
      </span>
      <span
        class="airbnbCost"
      >
        $230
      </span>
      <button
        class="airbnbBookingButton"
      >
        Book
      </button>
    </div>
  </div>
</div>
`;
